Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h

```diff
@@ -63,6 +69,7 @@ struct Polygon_soup_orienter
   Edge_map edges;             //< the set of edges of the input polygons
   Marked_edges marked_edges;  //< the set of singular edges or edges incident
                               //<   to non-compatible orientation polygons
+  const Visitor& visitor;
```

Comment:
Why retain a `const&` instead of a copy? That will complicate the use of non-const visitors.